src/evaluators/main_evaluator.py

```diff
@@ -0,0 +1,99 @@
+"""Main runner of the evaluation pipeline."""
+
+from pathlib import Path
+from typing import Any, Callable
+
+import yaml
+from tqdm import tqdm
+
+from src import os_utils, sim_utils
+from src.evaluators import loader, step_eval
+from src.icm import nd_model, torch_model
+
+
+def get_step_func(spreading_model_name: str) -> Callable:
+    if spreading_model_name == nd_model.FixedBudgetMICModel.__name__:
+        raise NotImplementedError(f"Pipeline for {spreading_model_name} is not yet ready!")
+    elif spreading_model_name == torch_model.TorchMICModel.__name__:
+        step_func = step_eval
+    else:
+        raise ValueError(f"Incorrect name of them model {spreading_model_name}")
+    print(f"Inferred step function as: {step_func.__name__}")
```

Comment:
(2) Why we are not using logging here?

Reply:
to be honest - I'm using massively `tqdm` here and if you mix it with `logging` you get neither progress bar nor pretty logs

that's why i decided to stick to `print`s

Reply:
Ok, may you add todo to check hydra for logging for this case (I'll do it, I just don't want to forget about it)

Reply:
done